src/monattack.cpp

```diff
@@ -3963,16 +3963,20 @@ bool mattack::lunge(monster *z)
     }
     body_part hit = target->get_random_body_part();
     int dam = rng(3, 7);
-
-    target->deal_damage( z, hit, damage_instance( DT_BASH, dam ) );
+    dam = target->deal_damage( z, hit, damage_instance( DT_BASH, dam ) ).total_damage();
+    if( dam > 0 ) {
+        target->add_msg_if_player(m_bad, _("Your %1$s is battered for %2$d damage!"), body_part_name(hit).c_str(), dam);
+    } else {
+        target->add_msg_player_or_npc( _("The %1$s lunges at your %2$s, but glances off your armor!"),
+                                    _("The %1$s lunges at <npcname>'s %2$s, but glances off armor!"),
```

Comment:
Glancing off suggests the effect is totally negated, but downing can still happen. Not sure how to say that better, though.
Lunging is already mentioned above. You could remove that message and integrate it here or keep it as 2 messages "lunges into" and then "but fails to deal damage". The former (one message per attack) would be preferable as most attacks look like that.
When a npc/monster avoids all damage a msg is printed, but no message is printed if npc/monster takes damage from the attack. It's OK to reveal how hard was a NPC/monster hit.